File: vbox-sdk/src/main/java/com/nexa/awesome/core/system/Camera/VirtualCameraService.java
package com.nexa.awesome.core.system.Camera;

import android.app.Service;
import android.content.Intent;
import android.graphics.SurfaceTexture;
import android.media.MediaPlayer;
import android.net.Uri;
import android.os.IBinder;
import android.util.Log;
import android.view.Surface;
import com.nexa.awesome.utils.compat.BuildCompat;

public class VirtualCameraService extends Service {
    private MediaPlayer mediaPlayer;

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        if (intent != null) {
            String videoPath = intent.getStringExtra("video_path");
            if (videoPath == null) {
                return START_NOT_STICKY;
            }
            boolean isNetwork = intent.getBooleanExtra("is_network", false);
            boolean enableAudio = intent.getBooleanExtra("enable_audio", false);

            stopMediaPlayer();
            setupVirtualCamera(videoPath, isNetwork, enableAudio);
        }
        return START_STICKY;
    }

    private void setupVirtualCamera(String videoPath, boolean isNetwork, boolean enableAudio) {
        try {
            if (isNetwork) {
                mediaPlayer = new MediaPlayer();
                mediaPlayer.setDataSource(videoPath);
                mediaPlayer.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
                    @Override
                    public void onPrepared(MediaPlayer mp) {
                        mp.start();
                    }
                });
                mediaPlayer.prepareAsync();
            } else {
                mediaPlayer = MediaPlayer.create(this, Uri.parse(videoPath));
            }

            if (mediaPlayer != null) {
                mediaPlayer.setLooping(true);
                mediaPlayer.setVolume(enableAudio ? 1f : 0f, enableAudio ? 1f : 0f);

                // Global Fake Surface
                SurfaceTexture surfaceTexture = new SurfaceTexture(10);
                Surface surface = new Surface(surfaceTexture);
                BuildCompat.surface = surface;

                mediaPlayer.setSurface(surface);
                mediaPlayer.start();
            }
        } catch (Exception e) {
            Log.e("VirtualCamera", "Error setting up virtual camera", e);
            stopSelf();
        }
    }

    private void stopMediaPlayer() {
        if (mediaPlayer != null) {
            mediaPlayer.stop();
            mediaPlayer.release();
            mediaPlayer = null;
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        stopMediaPlayer();
        BuildCompat.surface = null;
    }
}

File: app/src/main/java/com/nexa/awesome/camera/VirtualCameraService.kt
package com.nexa.awesome.camera

import android.app.Service
import android.net.Uri
import android.content.Intent
import android.graphics.SurfaceTexture
import android.hardware.camera2.CameraCharacteristics
import android.hardware.camera2.CameraDevice
import android.hardware.camera2.CameraManager
import android.media.MediaPlayer
import android.os.IBinder
import android.util.Log
import android.view.Surface
import java.util.concurrent.Executors

class VirtualCameraService : Service() {
    private lateinit var cameraManager: CameraManager
    private var mediaPlayer: MediaPlayer? = null
    private val executor = Executors.newSingleThreadExecutor()
    private var isRunning = false

    override fun onBind(intent: Intent?): IBinder? = null

    override fun onCreate() {
        super.onCreate()
        cameraManager = getSystemService(CAMERA_SERVICE) as CameraManager
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        intent?.let {
            val videoPath = it.getStringExtra("video_path") ?: return START_NOT_STICKY
            val isNetwork = it.getBooleanExtra("is_network", false)
            val enableAudio = it.getBooleanExtra("enable_audio", false)
            
            if (isRunning) {
                stopMediaPlayer()
            }
            
            executor.execute {
                try {
                    setupVirtualCamera(videoPath, isNetwork, enableAudio)
                    isRunning = true
                } catch (e: Exception) {
                    Log.e("VirtualCamera", "Error setting up virtual camera", e)
                    stopSelf()
                }
            }
        }
        return START_STICKY
    }

    private fun setupVirtualCamera(videoPath: String, isNetwork: Boolean, enableAudio: Boolean) {
        mediaPlayer = if (isNetwork) {
            MediaPlayer().apply {
                setDataSource(videoPath)
                setOnPreparedListener { start() }
                prepareAsync()
            }
        } else {
            MediaPlayer.create(this, Uri.parse(videoPath))
        }

        mediaPlayer?.apply {
            isLooping = true
            setVolume(if (enableAudio) 1.0f else 0.0f, if (enableAudio) 1.0f else 0.0f)
            
            // Create a surface for the virtual camera
            val surfaceTexture = SurfaceTexture(0)
            val surface = Surface(surfaceTexture)
            setSurface(surface)
            
            start()
        }
    }

    private fun stopMediaPlayer() {
        mediaPlayer?.let {
            it.stop()
            it.release()
        }
        mediaPlayer = null
        isRunning = false
    }

    override fun onDestroy() {
        super.onDestroy()
        stopMediaPlayer()
        executor.shutdown()
    }
}